Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -797,4 +797,41 @@ def test_custom_command
         cmd(cmd, {"PATH" => "#{path}#{File::PATH_SEPARATOR}#{ENV["PATH"]}"})
     end
   end
+
+  def test_search
+    formula_file = CoreTap.new.formula_dir/"testball.rb"
+    formula_file.write <<-EOS.undent
+      class Testball < Formula
+        desc "Some test"
+        url "https://example.com/testball-0.1.tar.gz"
+      end
+    EOS
+
+    desc_cache = HOMEBREW_CACHE.join("desc_cache.json")
+    refute_predicate desc_cache, :exist?, "Cached file should not exist"
+
+    assert_match "testball", cmd("search")
+    assert_match "testball", cmd("search", "testball")
+    assert_match "testball", cmd("search", "homebrew/homebrew-core/testball")
+    assert_match "testball", cmd("search", "--desc", "Some test")
+
+    flags = {
+      "macports" => "https://www.macports.org/ports.php?by=name&substr=testball",
+      "fink" => "http://pdb.finkproject.org/pdb/browse.php?summary=testball",
+      "debian" => "https://packages.debian.org/search?keywords=testball&searchon=names&suite=all&section=all",
+      "opensuse" => "https://software.opensuse.org/search?q=testball",
+      "fedora" => "https://admin.fedoraproject.org/pkgdb/packages/%2Atestball%2A/",
+      "ubuntu" => "http://packages.ubuntu.com/search?keywords=testball&searchon=names&suite=all&section=all"
```

Comment:
We've mostly settled on that trailing comma in the not to distant past, particularly in formulae, but it's of course good to be consistent in here. Though `%W[`-style arrays are also often possible in formulae and avoid the trailing comma discussion. :wink:

I don't think anyone is currently planning to produce such a document or wants to set aside time to do that. [Our RuboCop configuration](https://github.com/Homebrew/brew/blob/96cbce015e0b327bd7d30fc101cbd25452cd2fd8/Library/.rubocop.yml) documents our style to some degree, but other than that it's just word of mouth and looking at what has been done before.